app/pages/project/classify.jsx

```diff
@@ -148,15 +150,22 @@ export class ProjectClassifyPage extends React.Component {
   }
 
   shouldWorkflowAssignmentPrompt(nextProps) {
+    const { preferences, project } = this.props;
     // Only for Gravity Spy which is assigning workflows to logged in users
     if (nextProps.project.experimental_tools.indexOf('workflow assignment') > -1) {
       const assignedWorkflowID = nextProps.preferences &&
         nextProps.preferences.settings &&
         nextProps.preferences.settings.workflow_id;
-      const currentWorkflowID = this.props.preferences && this.props.preferences.preferences.selected_workflow;
+      const currentWorkflowID = preferences && preferences.preferences.selected_workflow;
```

Comment:
Current workflow ID will actually be `this.props.workflow.id`, which should be the same as user prefs but we can't assume that.